pagopa/io-app
# feat(Carta Giovani Nazionale): [#178358407,#178358013] Integrates merchants search API for CGN

## Short description
This PR introduces the API to search the Merchants for CGN disconunts.

Adds the sagas to handle API call and fixes the component displaying the list of Merchants

## How to test
From CGN detail screen, navigate to the list of merchants and then to the detail of a merchant. To test check that `cgn_merchants_v2` remote flag on `io-dev-api-server` is set to `false`

![Simulator Screen Recording - iPhone 11 - 2021-06-10 at 16 43 34](https://user-images.githubusercontent.com/3959405/121545696-09a21400-ca0b-11eb-82db-3f9ee3766218.gif)
